extensions/functions_datetime.yaml

```diff
@@ -374,6 +374,9 @@ scalar_functions:
             value: interval_year
           - name: y
             value: interval_year
+        options:
+          days_in_year:
+            values: [ YEAR_366DAYS, YEAR_365DAYS, YEAR_360DAYS ]
```

Comment:
I think we should hold off on introducing this option without thinking more broadly about this.

Additionally, this concept (days in year) is irrelevant to interval_day and interval_year. Neither of those mix days and years. It is really only relevant when you bring in interval_compound.